Assignment 13

task 1

Using Python and the scipy.stats library, perform a one-sample t-test to check if the average delivery time for Zomato orders in your city is significantly different from 30 minutes. Assume you have this sample data: [28, 32, 27, 35, 29, 31, 30, 33, 26, 34].

In [1]:
import numpy as np
from scipy import stats

delivery_times = [28, 32, 27, 35, 29, 31, 30, 33, 26, 34]
population_mean = 30  # the value we are testing against

sample_mean = np.mean(delivery_times)
t_stat, p_value = stats.ttest_1samp(delivery_times, population_mean)

alpha = 0.05

print(f"Sample mean delivery time = {round(sample_mean, 2)} minutes")
print(f"t-statistic = {round(t_stat, 4)}")
print(f"p-value = {round(p_value, 4)}")

if p_value < alpha:
    conclusion = "Reject H0 -> average delivery time IS significantly different from 30 minutes"
else:
    conclusion = "Fail to reject H0 -> no significant difference from 30 minutes"

print(f"Conclusion: {conclusion}")


Sample mean delivery time = 30.5 minutes
t-statistic = 0.5222
p-value = 0.6141
Conclusion: Fail to reject H0 -> no significant difference from 30 minutes


**Explanation:** A one-sample t-test compares the mean of a single sample against a known or hypothesized value — here, checking whether Zomato's average delivery time differs from 30 minutes. Using `scipy.stats.ttest_1samp(delivery_times, 30)`, the sample mean (30.5 minutes) is very close to 30, and the resulting p-value comes out well above 0.05, so we fail to reject the null hypothesis. This means there isn't enough evidence to say the average delivery time is significantly different from 30 minutes — the small variation we see could easily be due to random day-to-day fluctuation.

task 2

Collect or simulate two independent samples of daily step counts: one group for users before installing a fitness app (e.g., Google Fit) and one group after installation. Use Python to perform an independent (two-sample) t-test and determine if the average steps per day increased after using the app.

**Hint:** Use scipy.stats.ttest_ind().

In [2]:
import numpy as np
from scipy import stats

np.random.seed(42)  # for reproducible results

# Simulate two INDEPENDENT groups of users (different people in each group)
steps_before_app = np.random.normal(loc=6000, scale=800, size=25)   # users without the app
steps_after_app = np.random.normal(loc=6800, scale=800, size=25)    # different users, with the app

t_stat, p_value = stats.ttest_ind(steps_after_app, steps_before_app)
alpha = 0.05

print(f"Mean steps (before app, group 1) = {round(np.mean(steps_before_app), 2)}")
print(f"Mean steps (after app, group 2) = {round(np.mean(steps_after_app), 2)}")
print(f"t-statistic = {round(t_stat, 4)}")
print(f"p-value = {round(p_value, 4)}")

if p_value < alpha:
    conclusion = "Reject H0 -> average daily steps significantly increased after using the app"
else:
    conclusion = "Fail to reject H0 -> no significant increase in average daily steps"

print(f"Conclusion: {conclusion}")


Mean steps (before app, group 1) = 5869.19
Mean steps (after app, group 2) = 6570.05
t-statistic = 3.2908
p-value = 0.0019
Conclusion: Reject H0 -> average daily steps significantly increased after using the app


**Explanation:** An independent (two-sample) t-test is used when comparing the means of two separate, unrelated groups — here, one group of users measured before installing the fitness app and a *different* group of users measured after. `scipy.stats.ttest_ind()` compares these two independent samples. Since we deliberately simulated the "after" group with a higher true mean (6800 vs 6000 steps), the p-value comes out below 0.05, so we reject the null hypothesis and conclude the average daily steps did significantly increase after app installation.

task 3

Simulate paired data for 10 users showing their average screen time (in hours) before and after enabling 'Focus Mode' on their phones. Use Python to conduct a paired t-test and report whether 'Focus Mode' significantly reduced screen time.

**Hint:** Use scipy.stats.ttest_rel().

In [3]:
import numpy as np
from scipy import stats

np.random.seed(10)  # for reproducible results

# Simulate the SAME 10 users, measured before and after enabling Focus Mode (paired data)
screen_time_before = np.random.normal(loc=6.0, scale=1.0, size=10)
# Each user's "after" value is based on their own "before" value, minus some reduction
screen_time_after = screen_time_before - np.random.normal(loc=1.0, scale=0.4, size=10)

t_stat, p_value = stats.ttest_rel(screen_time_before, screen_time_after)
alpha = 0.05

print("Screen time before Focus Mode (hours):", np.round(screen_time_before, 2))
print("Screen time after Focus Mode (hours): ", np.round(screen_time_after, 2))
print(f"\nMean before = {round(np.mean(screen_time_before), 2)} hours")
print(f"Mean after  = {round(np.mean(screen_time_after), 2)} hours")
print(f"t-statistic = {round(t_stat, 4)}")
print(f"p-value = {round(p_value, 4)}")

if p_value < alpha:
    conclusion = "Reject H0 -> Focus Mode significantly reduced screen time"
else:
    conclusion = "Fail to reject H0 -> no significant reduction in screen time"

print(f"Conclusion: {conclusion}")


Screen time before Focus Mode (hours): [7.33 6.72 4.45 5.99 6.62 5.28 6.27 6.11 6.   5.83]
Screen time after Focus Mode (hours):  [6.16 5.23 3.84 4.58 5.53 4.1  5.72 5.05 4.41 5.26]

Mean before = 6.06 hours
Mean after  = 4.99 hours
t-statistic = 8.8502
p-value = 0.0
Conclusion: Reject H0 -> Focus Mode significantly reduced screen time


**Explanation:** A paired t-test is used when the same subjects are measured twice — once before and once after a treatment — as is the case here with the same 10 users' screen time before and after enabling Focus Mode. `scipy.stats.ttest_rel()` takes this pairing into account by looking at the differences within each user, rather than treating the two groups as unrelated. Since each user's "after" value was constructed to be lower than their own "before" value, the test finds a significant reduction (p-value below 0.05), supporting the conclusion that Focus Mode significantly reduced screen time.

task 4

Given this scenario: You want to compare the average ratings of two food delivery apps (Zomato vs Swiggy) based on 15 randomly selected user ratings for each app. Write Python code to perform the appropriate t-test, and explain why you chose that test.

In [4]:
import numpy as np
from scipy import stats

np.random.seed(5)  # for reproducible results

# 15 randomly selected, DIFFERENT users rated each app -> independent samples
zomato_ratings = np.random.normal(loc=4.2, scale=0.4, size=15)
swiggy_ratings = np.random.normal(loc=4.0, scale=0.4, size=15)

# Independent two-sample t-test, since the two sets of 15 ratings come from different users
t_stat, p_value = stats.ttest_ind(zomato_ratings, swiggy_ratings)
alpha = 0.05

print(f"Mean Zomato rating = {round(np.mean(zomato_ratings), 2)}")
print(f"Mean Swiggy rating = {round(np.mean(swiggy_ratings), 2)}")
print(f"t-statistic = {round(t_stat, 4)}")
print(f"p-value = {round(p_value, 4)}")

if p_value < alpha:
    conclusion = "Reject H0 -> the average ratings of Zomato and Swiggy are significantly different"
else:
    conclusion = "Fail to reject H0 -> no significant difference between the average ratings"

print(f"Conclusion: {conclusion}")


Mean Zomato rating = 4.19
Mean Swiggy rating = 3.99
t-statistic = 1.3757
p-value = 0.1798
Conclusion: Fail to reject H0 -> no significant difference between the average ratings


**Explanation:** This calls for an **independent (two-sample) t-test**, using `scipy.stats.ttest_ind()`, because the 15 ratings for Zomato and the 15 ratings for Swiggy come from different, unrelated users — there's no pairing between a specific Zomato rating and a specific Swiggy rating. This rules out a paired t-test (which needs the same subjects rating both apps) and a one-sample t-test (which compares one sample to a fixed value, not two samples to each other).

task 5

Use ChatGPT to help you write Python code that performs a one-sample t-test on a dataset of your choice (for example, average daily UPI transactions per user on PhonePe). Copy the code, run it on your data, and paste both the code and output in your submission.

In [5]:
# AI-generated code (from ChatGPT) for a one-sample t-test on PhonePe UPI transaction data
import numpy as np
from scipy import stats

np.random.seed(2)

# Simulated data: number of daily UPI transactions for 12 PhonePe users
daily_upi_transactions = np.random.normal(loc=5, scale=1.5, size=12).round().astype(int)
daily_upi_transactions = np.clip(daily_upi_transactions, 0, None)  # no negative transaction counts

claimed_average = 4  # PhonePe's claimed average daily transactions per user

t_stat, p_value = stats.ttest_1samp(daily_upi_transactions, claimed_average)
alpha = 0.05

print("Daily UPI transactions per user:", daily_upi_transactions)
print(f"Sample mean = {round(np.mean(daily_upi_transactions), 2)}")
print(f"Claimed average = {claimed_average}")
print(f"t-statistic = {round(t_stat, 4)}")
print(f"p-value = {round(p_value, 4)}")

if p_value < alpha:
    conclusion = "Reject H0 -> the true average differs significantly from the claimed value"
else:
    conclusion = "Fail to reject H0 -> no significant difference from the claimed value"

print(f"Conclusion: {conclusion}")


Daily UPI transactions per user: [4 5 2 7 2 4 6 3 3 4 6 8]
Sample mean = 4.5
Claimed average = 4
t-statistic = 0.8971
p-value = 0.3889
Conclusion: Fail to reject H0 -> no significant difference from the claimed value


**Explanation:** This follows the same one-sample t-test approach as Task 1, but applied to a different dataset: simulated daily UPI transaction counts for 12 PhonePe users, tested against a claimed average of 4 transactions per day. The output shows the actual simulated transaction counts, the sample mean, and the resulting t-statistic and p-value, letting us decide whether the observed data is consistent with PhonePe's claimed average or significantly different from it.